src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -138,8 +139,41 @@ internal static bool TryGetBestOverload(CheckTypesContext context, IErrorContain
 
                 var localWarnings = new LimitedSeverityErrorContainer(errors, DocumentErrorSeverity.Warning);
 
+                var checkTypeErrors = new ErrorContainer();
+
                 // Typecheck the invocation and infer the return type.
-                typeCheckSucceeded = maybeFunc.CheckTypes(context, args, argTypes, localWarnings, out returnType, out nodeToCoercedTypeMap);
+                typeCheckSucceeded = maybeFunc.CheckTypes(context, args, argTypes, checkTypeErrors, out returnType, out nodeToCoercedTypeMap);
+
+                if (!typeCheckSucceeded && checkTypeErrors.HasErrors() &&
+                    checkTypeErrors.GetErrors().All(error => context.NameResolver.Lookup(new DName(error.Tok.ToString()), out NameLookupInfo nameLookup) && nameLookup.Type.IsDeferred))
+                {
+                    typeCheckSucceeded = true;
+
+                    // If one of the arg was unknown and that generated error (e.g. type mismatch)
+                    // and return type could not be calculated and was error we assign it as unknown.
+                    // and if return type was Table, we assign it to be table of deferred, so operation like In can work.
+                    switch (returnType.Kind)
+                    {
+                        case DKind.Error:
+                            returnType = DType.Deferred;
+                            break;
+                        case DKind.Table:
+                            returnType = DType.EmptyRecord.Add(new TypedName(DType.Deferred, new DName(TexlFunction.ColumnName_ValueStr))).ToTable();
+                            break;
+                    }
+                }
+
+                // Adding back all the errors to local warnings
+                foreach (var error in checkTypeErrors.GetErrors())
+                {
+                    var isLookupSuccess = context.NameResolver.Lookup(new DName(error.Tok.ToString()), out var nameLookupInfo);
+
+                    // Only add error for nodes apart from Unknown type argument.
+                    if (!isLookupSuccess || nameLookupInfo.Type.IsDeferred)
+                    {
+                        localWarnings.EnsureError(error.Node, error.ErrorResourceKey);
```

Comment:
This could throw, what if the error is a format string? Looks like it loses the rest of the error context